nvfuser/__init__.py

```diff
@@ -202,10 +201,17 @@ def execute(
             self._setup_definition()
             self.definition()
             self._finalize_definition()
-            func_based_def = True
 
-        # If schedule is defined by child class, make a schedule for inputs
-        if func_based_def and (super(type(self), self).schedule != self.schedule):
+        # If schedule is defined by child class and schedule is not defined for
+        # inputs, make a schedule.
+        is_fusion_definition_child_class = (
+            type(self) != FusionDefinition
+        ) and issubclass(type(self), FusionDefinition)
+        defined_schedule = (
+            is_fusion_definition_child_class
```

Comment:
naive question, why do we need to check for `is_fusion_definition_child_class` here?

i.e. if we have a derived `schedule` method but went through the `execute` method here, wasn't that always true?

Reply:
We have the context manager option that won't create a schedule method. We
```python
with FusionDefinition() as fd:
  # create a fusion
out = fd.execute(inputs, device=device)
```

Reply:
I think we should simplify this check to `hasattr(self, "schedule") and callable(self.schedule)` and remove `schedule` from `FusionDefinition`

Reply:
Got ya. Thanks for showing that example.